python/pfs/drp/stella/lsf.py

```diff
@@ -0,0 +1,688 @@
+        """
+        self.values *= self._getOtherValues(other)
+        return self
+
+    def __itruediv__(self, other):
+        """Division in-place
+
+        Supports division by a scalar or an array.
+        """
+        self.values /= self._getOtherValues(other)
+        return self
+
+    def __reduce__(self):
+        """Pickle prescription"""
+        return self.__class__, (self.values, self.center, False)
+
+    def __repr__(self):
+        """Representation"""
+        return "%s(%s, %f)" % (self.__class__.__name__, self.values, self.center)
+
+    def normalization(self):
+        """Return the normalization"""
+        return np.sum(self.values)
+
+    def convolve(self, array):
+        """Convolve an array by the kernel
+
+        The convolved array has the same length as the input array in the usual
+        case that the array is longer than the kernel.
+
+        Note that the ends of the array will not be fully convolved.
+
+        Parameters
+        ----------
+        array : `numpy.ndarray`, floating-point
+            Array to convolve.
+
+        Raises
+        ------
+        RuntimeError
+            If the array is shorter than the kernel.
+        """
+        if len(array) < self.length:
+            raise RuntimeError("Array too short for convolution by this kernel")
+        if self.center != (self.length - 1)//2:
+            # Pad to put center of kernel in the middle, so we don't introduce a shift
+            left = self.center
+            right = self.length - self.center - 1
+            middle = max(left, right)
+            length = 2*middle + 1
+            kernel = np.zeros(length, dtype=self.values.dtype)
+            kernel[middle - left:middle + right + 1] = self.values
+        else:
+            kernel = self.values
+        return np.convolve(array, kernel, "same")
+
+    def toArray(self, length, center, interpolator=DEFAULT_INTERPOLATOR):
+        """Convert to an array at a nominated position
+
+        Parameters
+        ----------
+        length : `int`
+            Desired length of array.
+        center : `float`
+            Index position at which to insert the kernel.
+        interpolator : callable, optional
+            Interpolator factory. Calling this with arrays of ``x`` and ``y``
+            should produce an interpolator that can be called with an array of
+            ``x`` values at which to interpolate.
+
+        Returns
+        -------
+        array : `numpy.ndarray`, floating-point
+            Array containing inserted kernel.
+        """
+        return interpolator(self.indices + center, self.values)(np.arange(length))
+
+    def computeStdev(self):
+        """Compute standard deviation of kernel"""
+        centroid = np.sum((self.indices*self.values).astype(np.float64))
+        return np.sqrt(np.sum((self.values*(self.indices - centroid)**2).astype(np.float64)))
+
+    def interpolate(self, indices):
```

Comment:
What is the use case for this? Is it intended to provide an interface to properly over-sampled (and maybe Zernike-based) models?

In other words, is `Kernel1D` to be thought of as a particular, rather over-simplified, kernel or are you thinking of it as a real working class?